### ISSUE TITLE
c# - Renaming a field can cause it to rename the same word in the comments in other files in the same project that are completely unrelated

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#74577]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/c---Renaming-a-field-can-cause-it-to-re/10658745)._

---
I had a this property in a public class:
`private Dispatcher Dispatcher { get; }`
which I renamed:
`private Dispatcher m_dispatcher { get; }`

and it did stuff like this to other files:

```
/// <summary>
/// Wrapper class to provide a more reliable Dispatcher object
/// </summary>
	public static class DispatcherProvider
```
became

```
/// <summary>
/// Wrapper class to provide a more reliable m_dispatcher object
/// </summary>
	public static class DispatcherProvider
```

---
#### Feedback Bot on 5/14/2024, 10:34 PM: 

(private comment, text removed)
#### Feedback Bot on 5/21/2024, 11:23 AM: 

(private comment, text removed)
#### Shen Chen [MSFT] on 6/26/2024, 03:58 PM: 

(private comment, text removed)
#### Halliday, Sean on 6/28/2024, 03:50 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by Cosifne** [comment]
A few comments:
1. This behavior is directly linked to this line.https://github.com/dotnet/roslyn/blob/9a792ffa3c4d8716e508863a6f87eb8464c49f45/src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs#L710
2. Customer is complaining that, when they rename a private property, the comment in other class also get renamed.
We currently have many tests guarded the scenario when a public class gets renamed, all the string literal in comment should be updated. We might want to introduce another set of restrictions when the renaming symbol is private property/field/method, we should only update the string in comment within the class.
3. If user doesn't want that now, they can turn off 'Rename in Comment' to temp disable all the comment renaming.

**Comment by CyrusNajmabadi** [comment]
This is by design.  There is an option explicitly for this purpose.

**Comment by ATECoder** [comment]
> This is by design. There is an option explicitly for this purpose.

could you please point to this option.  thank you.

**Comment by CyrusNajmabadi** [comment]
![Image](https://github.com/user-attachments/assets/5442d4e8-c014-4d75-bfda-13f5e0b22ccc)

Choosing 'include comments' and 'include strings' controls if the same identifier will be updated if found in those respective locations.

**Comment by ATECoder** [comment]
Thank you! Found it. Elementary.  In the Rename dialog (double Ctrl+r), hit the down arrow to open the drop-down menu as displayed in your message above. These settings as well as the open status of the dialog persist across sessions of Visual Studio; an excellent design decision.

Seeing that these settings/features extend to all the projects (and documents (?)) of the solution, it would seem prudent to turn it off by default at least until the scope of renaming can also be set. In fact, it seems to me that a better approach might be to remove this altogether in favor of using search and replace to address comments and strings. My feeling is that Rename should apply only to code so I think this 'featire' report should be reopened.

In my case, I renamed Register to RegisterAndCertify. As a result comments about status and event registers were renamed in all projects of my solution, which took a bit of time to undo once I reialized it. 

Thanks again for your quick reply. We are so fortunate to have this community!
